src/main/java/com/cedrus/aeolion/kafkaspringpong/controller/SpringPongController.java

```diff
@@ -1,39 +1,60 @@
 package com.cedrus.aeolion.kafkaspringpong.controller;
 
-import com.cedrus.aeolion.kafkaspringpong.config.TopicConfig;
-import com.cedrus.aeolion.kafkaspringpong.kafka.SpringPongProducer;
-import com.cedrus.aeolion.kafkaspringpong.model.Message;
-import com.fasterxml.jackson.core.JsonProcessingException;
+import com.cedrus.aeolion.kafkaspringpong.model.*;
+import com.cedrus.aeolion.kafkaspringpong.services.BallAdderService;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
+@Slf4j
 @RestController
 public class SpringPongController {
-    private TopicConfig topicConfig;
-    private SpringPongProducer producer;
+    private final BallAdderService ballAdder;
 
     @Autowired
-    public SpringPongController(TopicConfig topicConfig, SpringPongProducer producer) {
-        this.topicConfig = topicConfig;
-        this.producer = producer;
+    public SpringPongController(BallAdderService ballAdder) {
+        this.ballAdder = ballAdder;
     }
 
-    @RequestMapping(value = "/ping", method = RequestMethod.POST)
-    @ResponseBody
-    public void producePing() throws JsonProcessingException {
-        createResponse(topicConfig.getPing());
-    }
+    @PostMapping(value = "/ball")
+    public ResponseEntity<SpringPongResponse> introduceBall(@RequestBody SpringPongRequest request) {
+        log.info("Received request to add ball.");
+        log.debug("Request = {}", request);
+
+        SpringPongResponse responseObj = new SpringPongResponse();
```

Comment:
final this please.  You're never going to change the reference, so it's a good idea.